src/Schema/Table.php

```diff
@@ -76,7 +76,15 @@ public function __construct(
         }
 
         foreach ($indexes as $idx) {
-            $this->_addIndex($idx);
+            $idxName = $idx->getName() === '' ? null : $idx->getName();
+
+            if ($idx->isPrimary()) {
+                $this->setPrimaryKey($idx->getColumns(), $idxName);
+            } elseif ($idx->isUnique()) {
+                $this->addUniqueIndex($idx->getColumns(), $idxName, $idx->getOptions());
+            } else {
+                $this->addIndex($idx->getColumns(), $idxName, $idx->getFlags(), $idx->getOptions());
+            }
```

Comment:
The `Index` class is not final which means that an app might use its own implementation. Your change would break this use-case.